app/controllers/items/new.js

```diff
@@ -131,6 +132,7 @@ export default GoodcityController.extend(
     },
 
     showPiecesInput: Ember.computed("codeId", function() {
+      this.clearSubformAttributes();
```

Comment:
An update inside a computed property is a big code smell.
Reading a boolean (`showPiecesInput `) should absolutely not modify any data.

You can call that function in your actions at the appropriate time instead :)

Reply:
or if it needs to react to a change, then better use an observer than a computed property